core/src/main/java/io/grpc/internal/CallCredentialsApplyingTransportFactory.java

```diff
@@ -126,5 +148,56 @@ public Attributes getTransportAttrs() {
         return delegate.newStream(method, headers, callOptions);
```

Comment:
Add a `if (pendingApplier.get() > 0)` check for shutdown in this code path as well?